docs/sphinx/developer/hpctoolkit.rst

```diff
@@ -0,0 +1,192 @@
+will be generated. After running the ``hpcstruct`` command, a ``executable.hpcstruct`` file will
+be generated. These two generated items will then become input used with ``hpcprof``, as shown
+above. If the file that you are analyzing is large or is using a lot of resources, then 
+``hpcprof-mpi`` could be better to use. The ``hpcprof-mpi`` command looks the same otherwise. The
+result of either ``hpcprof`` command is a generated "database" folder.
+
+At this point, we need the HPCViewer program to view the resulting database. The easiest way
+to engage with the HPCViewer is to do it locally. Therefore, we can tar up the generated 
+database folder and use ``scp`` to send it to a local machine. For example:
+
+.. code-block:: bash
+
+   $ tar -czvf database.tar hpctoolkit-executable-database-<job_id>/
+   $ scp username@lassen.llnl.gov:path/to/database.tar .
+   $ tar -xzvf database.tar
+
+From here, you can open the HPCViewer and select the untarred database folder we just sent over
+to be viewed. More information on how to use HPCViewer will be provided in the next section.
+
+Otherwise, using HPCViewer from the command line can be tricky since we need X11 forwarding. 
+In order to have X11 forwarding available, we have to ssh into the LC machine and compute node 
+a little differently:
+
+.. code-block:: bash
+
+   $ ssh -YC lassen
+   $ bsub -XF -W 60 -nnodes 1 -Is /bin/bash
+
+.. note::
+   If that doesn't work, you can also try ssh'ing into the LC machine with ``ssh -X -S none lassen``.
+
+From here we can run the same steps as before (listed at the top of this section). When we
+have generated the database folder, we will just call the hpcviewer program with 
+``hpcviewer hpctoolkit-executable-database-<job_id>/``. LLNL's documentation for HPCToolKit
+also provides an example command to use the ``hpctraceviewer`` tool.
+
+Using HPCViewer
+^^^^^^^^^^^^^^^
+
+Once you have your own version of HPCViewer locally, it is very easy to launch and open up the 
+database folder generated earlier. You can do this with just ``./hpcviewer`` and selecting the
+right database folder.
+
+For our use cases, we mostly used the "Hot Child" feature, but that is by no means the most
+valuable or most important feature that HPCViewer offers. To learn more about what HPCViewer
+can do, the instruction manual is `here <http://hpctoolkit.org/download/hpcviewer/>`_.
+
+.. note::
+   Depending on what's available on your local machine, you may have to download or update Java
+   in order to run hpcviewer. There are instructions `here <http://hpctoolkit.org/download/hpcviewer/>`_
+   for hpcviewer. You can get Java 8 from `here <https://www.oracle.com/java/technologies/javase/javase-jdk8-downloads.html>`_.
+
+
+Running with Hatchet
+^^^^^^^^^^^^^^^^^^^^
+
+`Hatchet <https://hatchet.readthedocs.io/en/latest/index.html>`_ is a tool that can better analyze performance metrics given from a variety of tools,
```

Comment:
I think it would be better to link to the hatchet github page, rather than straight to the docs.

Reply:
Done